test-suite/src/lib.rs

```diff
@@ -0,0 +1,140 @@
+use libtest_mimic::{Arguments, Trial};
+use reqwest::StatusCode;
+use serde::Deserialize;
+
+// Sample Test Case
+// test_name = "3.7-headers"
+// test_route = "3-headers"
+// expected_output = "Hello, world!"
+// expected_response_status = 200
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct TestCase {
+    pub test_name: String,
+    pub test_route: String,
+    pub expected_output: String,
+    pub expected_response_status: u16,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Runner {
+    pub name: String,
+    pub command: String,
+    pub port: u16,
+    pub args: Vec<String>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub enum ERunner {
```

Comment:
Now, I may be nitpicking, but I find this use of an enum very irregular. You have 3 cases, all of which store the same inner data, and the case itself isn't actually useful for anything (aside from serialization in the toml file; but that's not a good enough reason).

An enum is useful for "making illegal state unrepresentable" ([a very good explanation can be found here](https://fsharpforfunandprofit.com/posts/designing-with-types-making-illegal-states-unrepresentable/)). This enum is, indeed, creating *more* representable illegal states; for example, consider the case of `ERunner::Winter(Runner { name = "wrangler" })`.

You can easily remove this enum, which will also let you remove the trait as well. To deserialize from the toml, if you want to keep the `Winter =`, you could use a `Map<String, Runner>`. However, I also find that to be redundant; if the label is removed, the file can be reduced to:

```toml
[[matrix_runners]]
name = "winter"
...

[[matrix_runners]]
name = "wrangler"
...
```

which is more readable too.

To summarize, if you're not using an enum to store different data or as a way to create branching logic, stop and reconsider if you actually need an enum.

Reply:
I feel it'd be unfair not to mention that it would, indeed, be appropriate to use an enum here if, instead of deserializing, we were hardcoding the values, like so:

```rust
enum Runner {
  Winter,
  Wrangler,
}

impl Runner {
  fn port(&self) -> u16 {
    match self {
      Self::Winter => 8080,
      Self::Wrangler => 8787,
    }
  }
}

Reply:
If I do this then I need to hardcode the command, port, etc.

Reply:
I'm not suggesting we hardcode stuff, just saying that *if* we were going to hardcode, then an enum would have been a suitable option.

Reply:
Hardcoded it for now